internal/provider/resource_model.go

```diff
@@ -121,7 +121,17 @@ func resourceModelRead(ctx context.Context, d *schema.ResourceData, meta interfa
 	if err := d.Set("type", modelInfo.Type); err != nil {
 		return diag.FromErr(err)
 	}
+
 	// TODO: locate model config values form modelsAPI or other endpoint.
+	stateConfig := d.Get("config").(map[string]interface{})
+	for setting, _ := range stateConfig {
+		if value, exists := modelConfig[setting]; exists {
+			stateConfig[setting] = value
+		} else {
+			delete(stateConfig, setting)
+		}
+	}
+	d.Set("config", stateConfig)
```

Comment:
Yes, let's go with with this for now. I see there's a task to follow up and enhance the read to support better importing of config. 

We can use this in the meantime as it allows us to progress.